lib/widgets/login.dart

```diff
@@ -334,8 +335,8 @@ class _UsernamePasswordFormState extends State<_UsernamePasswordForm> {
     if (!usernameValid || !passwordValid) {
       return;
     }
-    final String username = usernameFieldState.value!;
-    final String password = passwordFieldState.value!;
+    final String username = usernameFieldState.value!.trim();
+    final String password = passwordFieldState.value!.trim();
```

Comment:
This is good for the username (aka the email, for Zulip realms where usernames are emails), but it seems less appropriate for the password. I think the usual practice is that if you put spaces in your password, then that's just part of the password like anything else.